compiler/qsc_passes/src/invert_block.rs

```diff
@@ -329,14 +329,17 @@ fn make_range_field(range_id: NodeId, field: PrimField) -> Expr {
 }
 
 fn make_array_index_range_reverse(core: &Table, arr_id: NodeId, arr_ty: &Ty) -> Expr {
+    let ns = core
```

Comment:
this vec can be a static array which is passed in by ref avoiding the allocation.